stl_text/ops/tokenizers/whitespace_tokenizer.py

```diff
@@ -1,44 +1,51 @@
-
+from torchtext.experimental.vocab import (
+    build_vocab_from_iterator,
+    load_vocab_from_file,
+    Vocab,
+    vocab,
+)
 
 class WhitespaceTokenizer(nn.Module):
-    def __init__(self, vocab_path: Optional[str] = None, trainable=False, speed: int = 0):
-        super(WhitespaceTokenizer, self).__init__()
-        self.trainable = trainable
-        self.speed = speed  # mock a real tokenizer: slowing down tokenization speed
+    r"""White space tokenizer and vocabulary builder
+
+    Args:
+        vocab_path Optional[str]: Path to vocabulary file. The file should contain tokens seperated by new line.
+        trainable (bool): Indicates whether to add tokens to dictionary on-the-fly
+
+    Example:
 
-        self.unknown = "unknown"
-        self.vocab: Dict[str, int] = {self.unknown: 0}
+        >>> from stl_text.ops.tokenizers import WhitespaceTokenizer
+        >>> whitespacetokenizer = WhitespaceTokenizer()
+        >>> jit_whitespacetokenizer = torch.jit.script(whitespacetokenizer)
+        >>> print(jit_whitespacetokenizer.code)
+        def forward(self,
+            text: str) -> List[int]:
+        _0 = (self.vocab).forward(torch.split(text, None, -1), )
+        return _0
+        
+    """
 
-        # load vocab
-        path_manager = PathManager()
+    def __init__(self, vocab_path: Optional[str] = None):
+        super(WhitespaceTokenizer, self).__init__()
+        self.unk_token = '<unk>'
         if vocab_path:
-            with path_manager.open(vocab_path, "r") as f:
-                for line in f.readlines():
-                    token = line.split()[0]
-                    self.vocab[token] = len(self.vocab)
+            path_manager = PathManager()
+            with path_manager.open(vocab_path, "r",encoding='utf-8') as f:
+                self.vocab = load_vocab_from_file(f,unk_token=self.unk_token)
+        else:
+            self.vocab = vocab(OrderedDict(),unk_token=self.unk_token)
+            
+        self.vocab = self.vocab.to_ivalue() #Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
+
 
     def forward(self, text: str) -> List[int]:
-        return self.numberize(self.tokenize(text))
-
-    def tokenize(self, text: str) -> List[str]:
-        # slowing down tokenization speed
-        count = 0
-        for i in range(self.speed):
-            count += 1
-        return text.split()
-
-    def numberize(self, tokens: List[str]) -> List[int]:
-        token_ids: List[int] = []
-        for token in tokens:
-            if token in self.vocab:
-                token_ids.append(self.vocab[token])
-            else:
-                if self.trainable:
-                    token_ids.append(len(self.vocab))
-                    self.vocab[token] = len(self.vocab)
-                else:
-                    token_ids.append(self.vocab[self.unknown])
-        return token_ids
+        return self.vocab(text.split())
+
+    @torch.jit.ignore
+    def build_vocab(self, data: List[str]) -> None:
```

Comment:
What is the reasoning behind making this a method of the tokenizer?

Reply:
> What is the reasoning behind making this a method of the tokenizer?

The original implementation of WhitespaceTokenizer included a trainable option that allow to build the vocab on the fly (https://github.com/hudeven/text/blob/9fc5b4962282d7e1dc8d5fa5eb3258f253981060/stl_text/ops/tokenizers/whitespace_tokenizer.py#L40). 

We discussed on possibility to separate "trainability" from querying (https://github.com/hudeven/text/pull/10#discussion_r563836230)

So as an initial attempt I just put this in separate function that fulfills the requirement of translation workflow. Although I do think that this is not an ideal solution to make it part of tokenizer and I would be happy to make API changes based on suggestions.